src/main/java/org/codebusters/audiogeek/preferencesagent/application/auth/UserAuthFilter.java

```diff
@@ -0,0 +1,48 @@
+package org.codebusters.audiogeek.preferencesagent.application.auth;
+
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.codebusters.audiogeek.preferencesagent.application.auth.token.HuellToken;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+import static java.util.Optional.ofNullable;
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+
+/**
+ * Spring Security filter that validates if given JWT is valid token and registers Principal in Security Context.
+ */
+@RequiredArgsConstructor
+@Slf4j
+class UserAuthFilter extends OncePerRequestFilter {
+    private final HuellToken huellToken;
+
+    @Override
+    protected void doFilterInternal(@NonNull HttpServletRequest request, @NonNull HttpServletResponse response, @NonNull FilterChain filterChain)
+            throws ServletException, IOException {
+        var ctx = SecurityContextHolder.getContext();
+        ctx.setAuthentication(null);
+        log.info("Authenticating request: {}", request.getRequestURL());
```

Comment:
1. So here is starting of auth. But where is information that user of given ID was authenticated correctly?
Please be sure you have always the complete of security logging: attempt of auth, then success or failure.

2. I think you could log here also some other details like the HTTP method, Host of the client and the User-Agent